src/main/kotlin/fi/aalto/cs/apluscourses/model/component/ScalaSdk.kt

```diff
@@ -0,0 +1,99 @@
+package fi.aalto.cs.apluscourses.model.component
+
+import com.intellij.openapi.application.writeAction
+import com.intellij.openapi.externalSystem.service.project.IdeModifiableModelsProviderImpl
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.OrderRootType
+import com.intellij.openapi.roots.libraries.LibraryTablesRegistrar
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFileManager
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.plugins.scala.project.external.ScalaSdkUtils
+import scala.Option
+import scala.jdk.javaapi.CollectionConverters
+import java.io.File
+import java.nio.file.Path
+
+class ScalaSdk(private val scalaVersion: String, project: Project) : Library(scalaVersion, project) {
+    override suspend fun downloadAndInstall(updating: Boolean) {
+        status = Status.LOADING
+        println("Downloading Scala SDK $scalaVersion")
+        val strippedScalaVersion = this.scalaVersion.substringAfter("scala-sdk-")
+        val zipUrl =
+            "https://github.com/lampepfl/dotty/releases/download/$strippedScalaVersion/scala3-$strippedScalaVersion.zip"
+        val sourcesUrl = "https://github.com/scala/scala3/archive/refs/tags/$strippedScalaVersion.zip"
+        val path = Path.of(project.basePath!!, ".libs")
+        downloadAndUnzipZip(zipUrl, path) // TODO only "scala3-$scalaVersion/lib"?
```

Comment:
What does this TODO mean?

Reply:
I was considering not including binaries when unzipping, but as they are very small it wouldn't make much difference